src/dbnode/client/errors.go

```diff
@@ -54,6 +54,35 @@ func IsBadRequestError(err error) bool {
 	return false
 }
 
+// IsResourceExhaustedError determines if the error is a resource exhausted error.
+func IsResourceExhaustedError(err error) bool {
```

Comment:
nit: Call this `IsResourceExhaustedErrorCode` to match the fact that it's matching on ErrorCode not ErrorType?